uiServices/src/testCaseServices.go

```diff
@@ -188,4 +188,48 @@ func getAllTestCases(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	rw.WriteHeader(http.StatusOK)
-}
\ No newline at end of file
+}
+
+func getTestCase(rw http.ResponseWriter, req *http.Request) {
+
+	testCasePathDir := getTestCaseHeader(req)
+	testCaseName := chi.URLParam(req, "testCaseName")
+
+	ValidateFileNameAndHeader(rw, req, testCasePathDir, testCaseName)
+
+	file, err := os.Open(fmt.Sprintf("%s%s.xml", testCasePathDir, testCaseName))
+	if err != nil {
+		logrus.Error("Test Case Name Not Found: " + testCasePathDir + testCaseName)
+		rw.WriteHeader(http.StatusNotFound)
+		return
+	}
+	defer file.Close()
+
+	var testCase testStrategies.TestDefinition
+
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Read File")
```

Comment:
I would suggest adding the error to the log as well. `logrus.Error("Cannot Read File`, err)`
These errors can sometime help you figure out what is going on.